### PR TITLE
Add json_library config with Jason default

### PR DESCRIPTION
**Pull request by chulkilee**
See #361

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -40,7 +40,8 @@ defmodule Exq.Mixfile do
     [
       {:elixir_uuid, ">= 1.2.0"},
       {:redix, ">= 0.8.1"},
-      {:poison, ">= 1.2.0 or ~> 2.0"},
+      {:poison, ">= 1.2.0 or ~> 2.0", optiona: true},
```
Comment (akira) [inline mix.exs]: Is this typo here, should be `optional`?

**Comment by akira** [comment]
Thanks for PR!!  Looks good except one typo in mix file.  👍

File: mix.exs (hunk 1 above)
Comment (akira) [reply mix.exs]: Will update this after merge as already looking at it.

**Comment by akira** [comment]
Merging.   Thanks again!

**Comment by abitdodgy** [comment]
Just curious, why does `poison` have to stay in the deps if handling is being switched to `Jason`? 

Wouldn't it make more sense to remove poison and leave it up to the user to add it to their own mix file if they want to use poison?

**Pull request merged**